snapcraft/plugins/maven.py

```diff
@@ -79,3 +111,19 @@ def build(self):
             wardir = os.path.join(self.installdir, 'war')
             os.makedirs(wardir, exist_ok=True)
             self.run(['cp', '-a'] + warfiles + [wardir])
+
+
+def _create_settings(settings_path):
+    proxy = urlparse(os.environ['http_proxy'])
+    os.makedirs(os.path.dirname(settings_path), exist_ok=True)
+    with open(settings_path, 'w') as f:
+        f.write(_MVN_SETTINGS.format(
+            proxy.hostname,
+            proxy.port,
+            _get_no_proxy_string()))
+
+
+def _get_no_proxy_string():
+    no_proxy = [k.strip() for k in
+                os.environ.get('no_proxy', 'localhost').split(',')]
```

Comment:
You are missing tests for when no_proxy has one value, and when it has multiple values.